src/items/queries/find-item-by-id.handler.ts

```diff
@@ -0,0 +1,22 @@
+import { QueryHandler } from '@nestjs/cqrs';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+
+import { FindItemByIdResult, FindItemByIdQuery } from './find-item-by-id.query';
+import { Item } from '../entities/item.entity';
+
+@QueryHandler(FindItemByIdQuery)
+export class FindItemByIdHandler {
+  constructor(
+    @InjectRepository(Item)
+    private readonly itemRepository: Repository<Item>,
+  ) {}
+
+  async execute(query: FindItemByIdQuery): Promise<FindItemByIdResult> {
+    const { id } = query;
+
+    const item: Item | null = await this.itemRepository.findOneBy({ id: id });
+
+    return { item: item };
```

Comment:
Dernière petite suggestion pour alléger le code:

```suggestion
    return this.itemRepository.findOneBy({ id });
```

Si tu le fais, faudra adapter l'autre query aussi et le contrôleur parce que le result va être un item, pas un object qui contient un item.

Tu peux juste faire `export type <Query_Name>Result = Item`.